### DISCUSSION TITLE
What is the best way to debug code?

### DISCUSSION THREAD

**Discussion by puzzir** [quaadgras/graphics.gd#26] [Q&A] [answered]
Thank you for the project, very interesting.
Please tell me how I can debug the code here? Is it possible to debug via IDE (VSCode, Goland)?

**Comment by Splizard** [comment, marked as answer]
I've used GDB in the past, the trick is to debug Godot and when things cross over to Go, GDB will continue to work as expected.

I usually will use a debug build of Godot so that I can investigate segfaults and the like.

**Reply by puzzir** [reply]
Thanks! 
From VSCode, debugging of go code works via the "GDB Debugger - Beyond" plugin. If I connect to the process with Godot.
For Goland, I haven't found how to set it up yet.

**Reply by ghost** [reply]
@Splizard can you provide some guidance on how to debug stuff like this?
```
================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.4.stable.official (4c311cbee68c0b66ff8ebb8b0defdd9979dd2a41)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /path/graphics/linux_amd64.so(+0x274340) [0x7fffe8c74340] (??:0)
-- END OF BACKTRACE --
================================================================
```

**Reply by Splizard** [reply]
@ring-c 

You can build Godot with:
`scons target=template_debug use_asan=yes`

Launch the project from the command line with GDB or a GDB frontend (ie. gdbgui)
`cd path/to/project/graphics && gdbgui godot.linuxbsd.template_debug.x86_64`

Try and work out where in the C++/Go code the crash occurred by inspecting the threads.

**Comment by stan220** [comment]
Hi, please share maybe someone was able to connect the debug in Goland? Or maybe give me a rough idea?

**Reply by Splizard** [reply]
What debugger does Goland use?

**Reply by stan220** [reply]
By Delve. But I didn't see a command(in delve) that could only compile without running.

As far as I understand, I have to compile a library with debug via dlv. I Should enable the debug in Goland and then launch Godot?

**Reply by Splizard** [reply]
Ah, delve [doesn't support](https://github.com/go-delve/delve/issues/873) c-shared build targets, you may have better luck with gdb.
